File: dotnet/progressive/Models/Progression.cs
using System;

namespace progressive.Models
{

    public class Progression
    {
        public int ProgressionID { get; set; }
        public int ModuleTaskID { get; set; }
        public int StudentID { get; set; }
        public int TaskAssessment { get; set; }
        public bool? Completed { get; set; }
        public DateTime DueDate { get; set; }
        public string Comments { get; set; }

        public ModuleTask Task { get; set; }
        public Student Student { get; set; }
    }
}

File: angular/progression/src/app/models/Progression.ts
import { Student } from './Student';
import { ModuleTask } from './ModuleTask';

export interface Progression {
    progressionID?: number | null;
    moduleTaskID: number;
    studentID: number;
    taskAssessment: number;
    completed: boolean;
    dueDate: Date | string;
    comments: string;

    task?: ModuleTask | null;
    student?: Student | null;
}
